src/core/tsi/gts_transport_security.c

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/tsi/gts_transport_security.h"
+#include <grpc/support/alloc.h>
+
+static gts_shared_resource* gts_resource = NULL;
```

Comment:
In the current gts implementation, we use a gpr_once variable to initialize the mutex, and use the mutex to control that there will be only a single thread to do all initialization works, including the memory allocation for pointer, create channel and cq e.t.c. 

I do not have a strong opinion regarding the use of struct or pointer, but I am not sure if it's necessary to have extra gts_init (which will be called in grpc_init) to initialize the mutex especially in a situation we may not know if we want to use gts or not.